src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Definition/RazorDefinitionEndpoint.cs

```diff
@@ -0,0 +1,144 @@
+            _foregroundDispatcher = foregroundDispatcher ?? throw new ArgumentNullException(nameof(foregroundDispatcher));
+            _documentResolver = documentResolver ?? throw new ArgumentNullException(nameof(documentResolver));
+            _componentSearchEngine = componentSearchEngine ?? throw new ArgumentNullException(nameof(componentSearchEngine));
+        }
+
+        public TextDocumentRegistrationOptions GetRegistrationOptions()
+        {
+            return new TextDocumentRegistrationOptions
+            {
+                DocumentSelector = RazorDefaults.Selector
+            };
+        }
+
+        public async Task<LocationOrLocationLinks> Handle(DefinitionParams request, CancellationToken cancellationToken)
+        {
+            if (request is null)
+            {
+                throw new ArgumentNullException(nameof(request));
+            }
+
+            var documentSnapshot = await Task.Factory.StartNew(() =>
+            {
+                var path = request.TextDocument.Uri.GetAbsoluteOrUNCPath();
+                _documentResolver.TryResolveDocument(path, out var documentSnapshot);
+                return documentSnapshot;
+            }, cancellationToken, TaskCreationOptions.None, _foregroundDispatcher.ForegroundScheduler).ConfigureAwait(false);
+
+            if (documentSnapshot is null)
+            {
+                return null;
+            }
+
+            if (!FileKinds.IsComponent(documentSnapshot.FileKind))
+            {
+                return null;
+            }
+
+            var codeDocument = await documentSnapshot.GetGeneratedOutputAsync().ConfigureAwait(false);
+            if (codeDocument.IsUnsupported())
+            {
+                return null;
+            }
+
+            var originTagHelperBinding = await GetOriginTagHelperBindingAsync(documentSnapshot, codeDocument, request.Position).ConfigureAwait(false);
+            if (originTagHelperBinding is null)
+            {
+                return null;
+            }
+
+            var originTagDescriptor = originTagHelperBinding.Descriptors.FirstOrDefault();
+            if (originTagDescriptor is null)
+            {
+                return null;
+            }
+
+            var originComponentDocumentSnapshot = await _componentSearchEngine.TryLocateComponentAsync(originTagDescriptor).ConfigureAwait(false);
+            if (originComponentDocumentSnapshot is null)
+            {
+                return null;
+            }
+
+            var originComponentUri = new UriBuilder
+            {
+                Path = originComponentDocumentSnapshot.FilePath,
+                Scheme = Uri.UriSchemeFile,
+                Host = string.Empty,
+            }.Uri;
+
+            return new LocationOrLocationLinks(new[]
+            {
+                new LocationOrLocationLink(new Location
+                {
+                    Uri = originComponentUri,
+                    Range = new Range(new Position(0, 0), new Position(0, 0)),
```

Comment:
This reads to me like we'll open the correct file but not go to any particular line. Is there a reason why this should always go to `0,0`?

Reply:
Not sure where else to go :)

Reply:
Ahh, because this GoToDef is only used if the FileType is a Component. Yeah, 0,0 is fine.